2.0/m_banprotect.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: Sebastian Nielsen */
+/* $ModAuthorMail: sebastian@sebbe.eu */
+/* $ModDesc: Prevents a lower rank from removing a listmode item set by a higher rank */
+/* $ModDepends: core 2.0 */
+
+#include "inspircd.h"
+
+class Banprotector
+{
+ public:
+	std::map<char, std::map<std::string, unsigned int>> banrank;
+	Banprotector() {}
+
+	void addrank(const char& modeparam, const std::string& banparam, const unsigned int& rank)
+	{
+		if (!banrank[modeparam][banparam])
+			banrank[modeparam][banparam] = rank;
+	}
+
+	bool checkrank(const char& modeparam, const std::string& banparam, const unsigned int& rank)
+	{
+		if (!banrank[modeparam][banparam])
+			return true;
+		if (banrank[modeparam][banparam] > rank)
+			return false;
+		return true;
+	}
+
+	void delrank(const char& modeparam, const std::string& banparam)
+	{
+		std::map<std::string, unsigned int> innerban = banrank[modeparam];
+		std::map<std::string, unsigned int>::iterator removeindex;
+		removeindex = innerban.find(banparam);
```

Comment:
See the same as above. Also, innerban might be `banrank.end()`  when using find if this doesn't exist.